vcpkg_ports/avro.patch

```diff
@@ -0,0 +1,377 @@
+diff --git a/lang/c/CMakeLists.txt b/lang/c/CMakeLists.txt
```

Comment:
Note: this is copied from `duckdb_avro`, without this we get errors about `avro_reader_reader` because the patch doesn't get applied to `avro-c` even though `duckdb_avro` is being built as a dependency

I imagine the automatic merging of `vcpkg.json` doesn't take `ports` into account